### ISSUE TITLE
Problem with id

### ISSUE THREAD

**Issue by tsteele** [evo-terren/Gremlin.Net.CosmosDb#45]
In the latest version 0.3.4.3-rc1 I have converted to using IVertex, so my vertex now includes Id:

`   public class Site : IVertex {
        public string Id { get; set; }
        public string SiteName { get; set; }
}`

However, when I create a site with a specified id, the id gets assigned a GUID and Id  appears as a separate property in the database.

How do set the DB id to a specified value?

**Comment by evo-terren** [comment]
I could very well be wrong, but I believe this is a limitation with how graphs work (at least in CosmosDB). I am having difficulty finding documentation to support this, however.

My team have set our ids to custom properties instead of using the built-in "id" field.

**Comment by evo-terren** [comment]
Just to be clear, did this used to work before the latest version?

**Comment by tsteele** [comment]
Yes it did work , The id field was included in VertexBase which you inherited.

I have managed to get it to work using:
`
[JsonProperty("id")]
public string Id { get; set; }
`

**Comment by tsteele** [comment]
This obviously forces it to serialise the Id field as id.

**Comment by BenjaBobs** [comment]
Do note though, that the native id property is not indexed the same way as custom properties, and if you're working with partitioned data, there might be significant performance to be gained by also storing the id in a custom property (my team uses the name idx.)

**Comment by tsteele** [comment]
Thanks for the replies, I think I'll go back to using the old version 0.3.4.2-rc1 due to the problems with this latest version, see #46 .  still think its a great lib.
